gm4_mob_curing/data/gm4_mob_curing/functions/potion_cleric/discount_curer.mcfunction

```diff
@@ -0,0 +1,13 @@
+# @s = potion trading cleric villager converted from witch.
+# run from initialise_potion_cleric
+
+# Builds the gossips array on storage to avoid a bug with modifying gossips
+data merge storage gm4_mob_curing:temp {Gossips:[{Type:"minor_positive",Value:20}]}
+data modify storage gm4_mob_curing:temp Gossips[0].Target set from entity @e[type=minecraft:item,limit=1,distance=..3,tag=gm4_gapple] Thrower
+data modify entity @s Gossips set from storage gm4:temp Gossips
+
+execute store result score $curer gm4_mob_cu_data run data get storage gm4_mob_curing:temp Gossips[0].Target[3] 0.001
+execute as @a[distance=..10] store result score @s gm4_mob_cu_data run data get entity @s UUID[3] 0.001
+execute as @a[distance=..10] if score @s gm4_mob_cu_data = $curer gm4_mob_cu_data run advancement grant @s only gm4:mob_curing
```

Comment:
This seems hacky. A function call to avoid the duplicate `@a` wouldn't give much of an improvement here, but why limit it to 10 blocks range in the first place? Shouldn't the curer also get that advancement if they dropped the item from a height of more than 10 blocks?

Reply:
How would you find and grant the advancement to the thrower?  This fake curing is not defined under the typical Zombie Doctor curing advancement, and you cannot select an entity based on a score value.  You also cannot advancement grant <score value>, so even though I have the UUID of the player, it doesn't seem like I can access them any other way.  The distance check is for lag reduction.

Reply:
The item has a NB tag which stores the UUID of the thrower.